### PR TITLE
Reduce the number of declaration for Bio-Formats dependencies

### PR DESCRIPTION
**Pull request by sbesson**
formats-bsd, netcdf and ome-xml are transitive dependencies of formats-gpl. This should reduce the number of places to update when bumping Bio-Formats. The udunits version was not removed as it is different from the one declared by edu.ucar:netcdf:4.3.22

This is not strictly necessary for the upcoming Bio-Formats work but mostly opening the discussion about reducing the declaration of our dependencies. Importantly these changes assume we do not want to override transitive dependencies individually.

If the statement above is not true i.e. we foresee a case for bumping `ome-xml` totally independently of `formats-{bsd,gpl}` then the current layout should be preserved and this PR can be closed. Note in that case, I am unclear whether the declaration order matters for the dependency resolution.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -29,11 +29,9 @@ dependencies {
     api("org.hibernate:hibernate-search:3.4.2.Final")
 
     // Our libraries
-    api("org.openmicroscopy:ome-xml:5.6.3")
-    api("ome:formats-bsd:5.9.2") {
+    api("ome:formats-gpl:5.9.2") {
```
Comment (jburel) [inline build.gradle]: Bump to 6.0.0
I tested locally with B-F 6 and everything works smoothly so far

Comment (mtbc) [reply build.gradle]: We've had 6.0.1 out awhile, could go straight to that.

Comment (jburel) [reply build.gradle]: yes I was mainly testing the major version bump.

Comment (sbesson) [reply build.gradle]: What do you want to do here? Should I list that for review now or would you rather work on top of it for the BF 6 upgrade?

Comment (jburel) [reply build.gradle]: I tested your change and replaced 5.9.2 by 6. Let's work on top for BF upgrade

**Comment by jburel** [comment]
I have pushed a commit bumping to 6.0.1 so it can be included in next week build

**Comment by mtbc** [comment]
--exclude to see if it fixes west-ci.

**Comment by jburel** [comment]
The error
```
'java.lang.NoSuchMethodError: com.google.common.collect.Sets$SetView.iterator()Lcom/google/common/collect/UnmodifiableIterator;\n\tat ome.services.graphs.GraphTraversal.reviewObject(GraphTraversal.java:1212)
```
is no longer there with the latest change.

**Comment by jburel** [comment]
https://web-proxy.openmicroscopy.org/west-ci/job/OMERO-test-integration/358/ is green

**Comment by sbesson** [comment]
I am still confused why 2607fff and 47e5c69 are required at all. From da01fbc and as per the [Gradle dependency resolution](https://docs.gradle.org/5.2.1/userguide/troubleshooting_dependency_resolution.html#sub:version_conflicts), it looks like `guava:25.1` is resolved:

```
sbesson@ls30630:omero-model (bf_dependency) $ ./gradlew -q dependencies | grep gua
|    |    |    +--- com.google.guava:guava:25.1-jre
|    |    +--- com.google.guava:guava:17.0 -> 25.1-jre (*)
|    |    +--- com.google.guava:guava:17.0 -> 25.1-jre (*)
|    |    |    +--- com.google.guava:guava:25.1-jre
|    |    +--- com.google.guava:guava:17.0 -> 25.1-jre (*)
|    |    +--- com.google.guava:guava:17.0 -> 25.1-jre (*)
|    |    |    +--- com.google.guava:guava:25.1-jre
|    |    +--- com.google.guava:guava:17.0 -> 25.1-jre (*)
|    |    +--- com.google.guava:guava:17.0 -> 25.1-jre (*)
|    |    |    +--- com.google.guava:guava:25.1-jre
|    |    +--- com.google.guava:guava:17.0 -> 25.1-jre (*)
|    |    +--- com.google.guava:guava:17.0 -> 25.1-jre (*)
|    |    |    +--- com.google.guava:guava:25.1-jre
|    |    +--- com.google.guava:guava:17.0 -> 25.1-jre (*)
|    |    +--- com.google.guava:guava:17.0 -> 25.1-jre (*)
```

Does this mean that `omero-build` effectively was compiling with Guava 25.1 but the latests commits are required to enforce that Ivy will later pick the same version? There are certainly a few [threads](https://www.google.com/search?q=com.google.common.collect.Sets$SetView.iterator+site:stackoverflow.com&sa=X&ved=2ahUKEwid5-Dq8MLhAhWRVBUIHVeBBpAQrQIoBDADegQIBhAM&biw=1440&bih=664) reporting similar exception due to mismatching Guava versions/classpath issues.

**Comment by jburel** [comment]
While doing some testing, @will-moore noticed an issue with pyramid creation for large jpeg, png.
@dgault is currently looking into it.

**Comment by jburel** [comment]
Excluding for some tests
--exclude

**Comment by jburel** [comment]
This PR prevents the integration of omero-java-gateway due to guava problem. This might require a workaround at the ivy level

**Comment by jburel** [comment]
Confirmed by @will-moore that without this PR, the newly imported pyramid did not have any problem. see 5.5.0 testing sheet

**Comment by jburel** [comment]
--exclude

**Comment by jburel** [comment]
--exclude

**Comment by jburel** [comment]
I pushed away 47e5c69 and 2607fff to test the gateway integration
This is really problematic

**Comment by jburel** [comment]
As it stands we cannot include omero-java-gateway and BF 6

**Comment by jburel** [comment]
--exclude

**Comment by jburel** [comment]
--exclude

**Comment by jburel** [comment]
Closing this PR. https://github.com/ome/omero-model/pull/23/commits/f12e468ea6500c5cdcf743396f9740ae3594736f has been merged

**Pull request closed without merging**